// angular-dashjs-player/bundles/angular-dashjs-player.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('dashjs')) :
    typeof define === 'function' && define.amd ? define('angular-dashjs-player', ['exports', '@angular/core', 'dashjs'], factory) :
    (factory((global['angular-dashjs-player'] = {}),global.ng.core,null));
}(this, (function (exports,i0,dashjs) { 'use strict';

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var DashjsPlayerService = (function () {
        function DashjsPlayerService() {
        }
        DashjsPlayerService.decorators = [
            { type: i0.Injectable, args: [{
                        providedIn: 'root'
                    },] },
        ];
        /** @nocollapse */
        DashjsPlayerService.ctorParameters = function () { return []; };
        /** @nocollapse */ DashjsPlayerService.ngInjectableDef = i0.defineInjectable({ factory: function DashjsPlayerService_Factory() { return new DashjsPlayerService(); }, token: DashjsPlayerService, providedIn: "root" });
        return DashjsPlayerService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var DashjsPlayerComponent = (function () {
        function DashjsPlayerComponent() {
            this.autoload = true;
        }
        Object.defineProperty(DashjsPlayerComponent.prototype, "src", {
            set: /**
             * @param {?} value
             * @return {?}
             */ function (value) {
                this._src = value;
                if (this.player && this.player.isPaused()) {
                    this.play();
                }
            },
            enumerable: true,
            configurable: true
        });
        /**
         * @return {?}
         */
        DashjsPlayerComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
                this.player = dashjs.MediaPlayer().create();
                this.player.initialize(this.videoPlayer.nativeElement, this._src, this.autoload);
            };
        /**
         * @return {?}
         */
        DashjsPlayerComponent.prototype.ngOnDestroy = /**
         * @return {?}
         */
            function () {
                this.player.reset();
            };
        /**
         * @return {?}
         */
        DashjsPlayerComponent.prototype.getPlayer = /**
         * @return {?}
         */
            function () {
                return this.player;
            };
        /**
         * @return {?}
         */
        DashjsPlayerComponent.prototype.play = /**
         * @return {?}
         */
            function () {
                this.player.play();
            };
        /**
         * @return {?}
         */
        DashjsPlayerComponent.prototype.pause = /**
         * @return {?}
         */
            function () {
                this.player.pause();
            };
        DashjsPlayerComponent.decorators = [
            { type: i0.Component, args: [{
                        selector: 'dashjs-player',
                        template: "<video #videoPlayer controls></video>\n",
                        styles: ["video{width:100%;height:100%}"]
                    },] },
        ];
        /** @nocollapse */
        DashjsPlayerComponent.ctorParameters = function () { return []; };
        DashjsPlayerComponent.propDecorators = {
            videoPlayer: [{ type: i0.ViewChild, args: ['videoPlayer',] }],
            autoload: [{ type: i0.Input }],
            src: [{ type: i0.Input }]
        };
        return DashjsPlayerComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var DashjsPlayerModule = (function () {
        function DashjsPlayerModule() {
        }
        DashjsPlayerModule.decorators = [
            { type: i0.NgModule, args: [{
                        imports: [],
                        declarations: [DashjsPlayerComponent],
                        exports: [DashjsPlayerComponent]
                    },] },
        ];
        return DashjsPlayerModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */

    exports.DashjsPlayerService = DashjsPlayerService;
    exports.DashjsPlayerComponent = DashjsPlayerComponent;
    exports.DashjsPlayerModule = DashjsPlayerModule;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
