// angular-dashjs-player/fesm5/angular-dashjs-player.js
import { Injectable, Component, Input, ViewChild, NgModule, defineInjectable } from '@angular/core';
import { MediaPlayer } from 'dashjs';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var DashjsPlayerService = /** @class */ (function () {
    function DashjsPlayerService() {
    }
    DashjsPlayerService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] },
    ];
    /** @nocollapse */
    DashjsPlayerService.ctorParameters = function () { return []; };
    /** @nocollapse */ DashjsPlayerService.ngInjectableDef = defineInjectable({ factory: function DashjsPlayerService_Factory() { return new DashjsPlayerService(); }, token: DashjsPlayerService, providedIn: "root" });
    return DashjsPlayerService;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var DashjsPlayerComponent = /** @class */ (function () {
    function DashjsPlayerComponent() {
        this.autoload = true;
    }
    Object.defineProperty(DashjsPlayerComponent.prototype, "src", {
        set: /**
         * @param {?} value
         * @return {?}
         */
        function (value) {
            this._src = value;
            if (this.player && this.player.isPaused()) {
                this.play();
            }
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    DashjsPlayerComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.player = MediaPlayer().create();
        this.player.initialize(this.videoPlayer.nativeElement, this._src, this.autoload);
    };
    /**
     * @return {?}
     */
    DashjsPlayerComponent.prototype.ngOnDestroy = /**
     * @return {?}
     */
    function () {
        this.player.reset();
    };
    /**
     * @return {?}
     */
    DashjsPlayerComponent.prototype.getPlayer = /**
     * @return {?}
     */
    function () {
        return this.player;
    };
    /**
     * @return {?}
     */
    DashjsPlayerComponent.prototype.play = /**
     * @return {?}
     */
    function () {
        this.player.play();
    };
    /**
     * @return {?}
     */
    DashjsPlayerComponent.prototype.pause = /**
     * @return {?}
     */
    function () {
        this.player.pause();
    };
    DashjsPlayerComponent.decorators = [
        { type: Component, args: [{
                    selector: 'dashjs-player',
                    template: "<video #videoPlayer controls></video>\n",
                    styles: ["video{width:100%;height:100%}"]
                },] },
    ];
    /** @nocollapse */
    DashjsPlayerComponent.ctorParameters = function () { return []; };
    DashjsPlayerComponent.propDecorators = {
        videoPlayer: [{ type: ViewChild, args: ['videoPlayer',] }],
        autoload: [{ type: Input }],
        src: [{ type: Input }]
    };
    return DashjsPlayerComponent;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var DashjsPlayerModule = /** @class */ (function () {
    function DashjsPlayerModule() {
    }
    DashjsPlayerModule.decorators = [
        { type: NgModule, args: [{
                    imports: [],
                    declarations: [DashjsPlayerComponent],
                    exports: [DashjsPlayerComponent]
                },] },
    ];
    return DashjsPlayerModule;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

export { DashjsPlayerService, DashjsPlayerComponent, DashjsPlayerModule };
