// angular-dashjs-player/fesm2015/angular-dashjs-player.js
import { Injectable, Component, Input, ViewChild, NgModule, defineInjectable } from '@angular/core';
import { MediaPlayer } from 'dashjs';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class DashjsPlayerService {
    constructor() { }
}
DashjsPlayerService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] },
];
/** @nocollapse */
DashjsPlayerService.ctorParameters = () => [];
/** @nocollapse */ DashjsPlayerService.ngInjectableDef = defineInjectable({ factory: function DashjsPlayerService_Factory() { return new DashjsPlayerService(); }, token: DashjsPlayerService, providedIn: "root" });

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class DashjsPlayerComponent {
    constructor() {
        this.autoload = true;
    }
    /**
     * @param {?} value
     * @return {?}
     */
    set src(value) {
        this._src = value;
        if (this.player && this.player.isPaused()) {
            this.play();
        }
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.player = MediaPlayer().create();
        this.player.initialize(this.videoPlayer.nativeElement, this._src, this.autoload);
    }
    /**
     * @return {?}
     */
    ngOnDestroy() {
        this.player.reset();
    }
    /**
     * @return {?}
     */
    getPlayer() {
        return this.player;
    }
    /**
     * @return {?}
     */
    play() {
        this.player.play();
    }
    /**
     * @return {?}
     */
    pause() {
        this.player.pause();
    }
}
DashjsPlayerComponent.decorators = [
    { type: Component, args: [{
                selector: 'dashjs-player',
                template: `<video #videoPlayer controls></video>
`,
                styles: [`video{width:100%;height:100%}`]
            },] },
];
/** @nocollapse */
DashjsPlayerComponent.ctorParameters = () => [];
DashjsPlayerComponent.propDecorators = {
    videoPlayer: [{ type: ViewChild, args: ['videoPlayer',] }],
    autoload: [{ type: Input }],
    src: [{ type: Input }]
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class DashjsPlayerModule {
}
DashjsPlayerModule.decorators = [
    { type: NgModule, args: [{
                imports: [],
                declarations: [DashjsPlayerComponent],
                exports: [DashjsPlayerComponent]
            },] },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

export { DashjsPlayerService, DashjsPlayerComponent, DashjsPlayerModule };
