// angular-dashjs-player/esm5/lib/dashjs-player.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component, ElementRef, Input, ViewChild } from '@angular/core';
import { MediaPlayer } from 'dashjs';
var DashjsPlayerComponent = /** @class */ (function () {
    function DashjsPlayerComponent() {
        this.autoload = true;
    }
    Object.defineProperty(DashjsPlayerComponent.prototype, "src", {
        set: /**
         * @param {?} value
         * @return {?}
         */
        function (value) {
            this._src = value;
            if (this.player && this.player.isPaused()) {
                this.play();
            }
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    DashjsPlayerComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.player = MediaPlayer().create();
        this.player.initialize(this.videoPlayer.nativeElement, this._src, this.autoload);
    };
    /**
     * @return {?}
     */
    DashjsPlayerComponent.prototype.ngOnDestroy = /**
     * @return {?}
     */
    function () {
        this.player.reset();
    };
    /**
     * @return {?}
     */
    DashjsPlayerComponent.prototype.getPlayer = /**
     * @return {?}
     */
    function () {
        return this.player;
    };
    /**
     * @return {?}
     */
    DashjsPlayerComponent.prototype.play = /**
     * @return {?}
     */
    function () {
        this.player.play();
    };
    /**
     * @return {?}
     */
    DashjsPlayerComponent.prototype.pause = /**
     * @return {?}
     */
    function () {
        this.player.pause();
    };
    DashjsPlayerComponent.decorators = [
        { type: Component, args: [{
                    selector: 'dashjs-player',
                    template: "<video #videoPlayer controls></video>\n",
                    styles: ["video{width:100%;height:100%}"]
                },] },
    ];
    /** @nocollapse */
    DashjsPlayerComponent.ctorParameters = function () { return []; };
    DashjsPlayerComponent.propDecorators = {
        videoPlayer: [{ type: ViewChild, args: ['videoPlayer',] }],
        autoload: [{ type: Input }],
        src: [{ type: Input }]
    };
    return DashjsPlayerComponent;
}());
export { DashjsPlayerComponent };
function DashjsPlayerComponent_tsickle_Closure_declarations() {
    /** @type {?} */
    DashjsPlayerComponent.prototype.player;
    /** @type {?} */
    DashjsPlayerComponent.prototype._src;
    /** @type {?} */
    DashjsPlayerComponent.prototype.videoPlayer;
    /** @type {?} */
    DashjsPlayerComponent.prototype.autoload;
}
