// angular-dashjs-player/esm2015/lib/dashjs-player.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component, ElementRef, Input, ViewChild } from '@angular/core';
import { MediaPlayer } from 'dashjs';
export class DashjsPlayerComponent {
    constructor() {
        this.autoload = true;
    }
    /**
     * @param {?} value
     * @return {?}
     */
    set src(value) {
        this._src = value;
        if (this.player && this.player.isPaused()) {
            this.play();
        }
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.player = MediaPlayer().create();
        this.player.initialize(this.videoPlayer.nativeElement, this._src, this.autoload);
    }
    /**
     * @return {?}
     */
    ngOnDestroy() {
        this.player.reset();
    }
    /**
     * @return {?}
     */
    getPlayer() {
        return this.player;
    }
    /**
     * @return {?}
     */
    play() {
        this.player.play();
    }
    /**
     * @return {?}
     */
    pause() {
        this.player.pause();
    }
}
DashjsPlayerComponent.decorators = [
    { type: Component, args: [{
                selector: 'dashjs-player',
                template: `<video #videoPlayer controls></video>
`,
                styles: [`video{width:100%;height:100%}`]
            },] },
];
/** @nocollapse */
DashjsPlayerComponent.ctorParameters = () => [];
DashjsPlayerComponent.propDecorators = {
    videoPlayer: [{ type: ViewChild, args: ['videoPlayer',] }],
    autoload: [{ type: Input }],
    src: [{ type: Input }]
};
function DashjsPlayerComponent_tsickle_Closure_declarations() {
    /** @type {?} */
    DashjsPlayerComponent.prototype.player;
    /** @type {?} */
    DashjsPlayerComponent.prototype._src;
    /** @type {?} */
    DashjsPlayerComponent.prototype.videoPlayer;
    /** @type {?} */
    DashjsPlayerComponent.prototype.autoload;
}
